### PR TITLE
New triggers and actions ticktick

### PR DESCRIPTION
**Pull request by bush4u**

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible

File: components/ticktick/actions/update-task/update-task.mjs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      optional: true,
+    };
+    props.dueDate = {
+      type: "string",
+      label: "Due date",
+      description: "Due date and time in \"yyyy-MM-dd'T'HH:mm:ssZ\" format. Example : \"2019-11-13T03:00:00+0000\"",
+      optional: true,
+    };
+    return props;
+  },
```
Comment (michelle0927) [inline components/ticktick/actions/update-task/update-task.mjs:51]: Do these need to be `additionalProps`? Couldn't they be listed with the main props?

File: components/ticktick/common/constants.mjs [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  BASE_URL: "api.ticktick.com",
+  ROOT_PATH: "/open",
+  VERSION_PATH: "/v1",
+  PAGE_SIZE: 100,
+  retriableStatusCodes: [
+    408,
+    429,
+    500,
+  ],
+  DB_LAST_DATE_CHECK: "DB_LAST_DATE_CHECK",
```
Comment (michelle0927) [inline components/ticktick/common/constants.mjs:12]: I don't see `PAGE_SIZE`, `retriableStatusCodes`, or `DB_LAST_DATE_CHECK` being used anywhere.
```suggestion
```

File: components/ticktick/common/utils.mjs [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    11:45 AM
+    11:45 am
+    11:45am
+*/
+const isValidTime = (timeString) => {
+  const regex = /([0-9]|0[0-9]|1[0-9]|2[0-3]):([0-5][0-9])\s*([AaPp][Mm])$/;
+  return timeString
+    ? timeString.match(regex)
+    : false;
+};
```
Comment (michelle0927) [inline components/ticktick/common/utils.mjs:62]: Were `isValidDate` and `isValidTime` meant to be used to check the `startDate` and `dueDate`? I don't see them being used anywhere.

**Review by michelle0927 — CHANGES_REQUESTED** [review]

File: components/ticktick/actions/update-task/update-task.mjs (hunk 1 above)
Comment (bush4u) [reply components/ticktick/actions/update-task/update-task.mjs:51]: I don't think they should be listed. I am meant to include a control there

**Pull request merged**